qiskit/visualization/timeline/interface.py

```diff
@@ -25,21 +25,26 @@
 from qiskit.exceptions import MissingOptionalLibraryError
 from qiskit.visualization.exceptions import VisualizationError
 from qiskit.visualization.timeline import types, core, stylesheet
+from qiskit.utils import deprecate_arg
 
 
+@deprecate_arg("show_idle", new_alias="idle_wires", since="1.1.0")
+@deprecate_arg("show_barriers", new_alias="plot_barriers", since="1.1.0")
 def draw(
     program: circuit.QuantumCircuit,
     style: Optional[Dict[str, Any]] = None,
     time_range: Tuple[int, int] = None,
     disable_bits: List[types.Bits] = None,
     show_clbits: Optional[bool] = None,
-    show_idle: Optional[bool] = None,
-    show_barriers: Optional[bool] = None,
+    idle_wires: Optional[bool] = None,
+    plot_barriers: Optional[bool] = None,
     show_delays: Optional[bool] = None,
     show_labels: bool = True,
     plotter: Optional[str] = types.Plotter.MPL.value,
     axis: Optional[Any] = None,
     filename: Optional[str] = None,
+    show_idle: Optional[bool] = None,
+    show_barriers: Optional[bool] = None,
```

Comment:
We should probably make these keyword only so people can't accidentally set them positionally:

```suggestion
    filename: Optional[str] = None,
    *,
    show_idle: Optional[bool] = None,
    show_barriers: Optional[bool] = None,
```